Client/Frontend/Browser/Playlist/Controllers/PlaylistListViewController+TableViewDelegate.swift

```diff
@@ -32,6 +29,87 @@ extension PlaylistListViewController: UITableViewDelegate {
+                    
+                    alert.addAction(UIAlertAction(title: Strings.OKString, style: .default, handler: { [unowned self] _ in
+                        PlaylistManager.shared.download(item: item)
+                        self.tableView.reloadRows(at: [indexPath], with: .automatic)
+                    }))
+                    
+                    alert.addAction(UIAlertAction(title: Strings.cancelButtonTitle, style: .cancel, handler: nil))
+                    self.present(alert, animated: true, completion: nil)
+                } else {
+                    PlaylistManager.shared.download(item: item)
+                    tableView.reloadRows(at: [indexPath], with: .automatic)
+                }
+            case .downloaded:
+                let style: UIAlertController.Style = UIDevice.current.userInterfaceIdiom == .pad ? .alert : .actionSheet
+                let alert = UIAlertController(
+                    title: Strings.PlayList.removePlaylistOfflineDataAlertTitle, message: Strings.PlayList.removePlaylistOfflineDataAlertMessage, preferredStyle: style)
+                
+                alert.addAction(UIAlertAction(title: Strings.PlayList.removeActionButtonTitle, style: .destructive, handler: { [unowned self] _ in
+                    _ = PlaylistManager.shared.deleteCache(item: item)
+                    self.tableView.reloadRows(at: [indexPath], with: .automatic)
+                }))
+                
+                alert.addAction(UIAlertAction(title: Strings.cancelButtonTitle, style: .cancel, handler: nil))
+                self.present(alert, animated: true, completion: nil)
+        }
+    }
+    
+    func deleteItem(_ item: PlaylistInfo, indexPath: IndexPath) {
+        let style: UIAlertController.Style = UIDevice.current.userInterfaceIdiom == .pad ? .alert : .actionSheet
+        let alert = UIAlertController(
+            title: Strings.PlayList.removePlaylistVideoAlertTitle, message: Strings.PlayList.removePlaylistVideoAlertMessage, preferredStyle: style)
+        
+        alert.addAction(UIAlertAction(title: Strings.PlayList.removeActionButtonTitle, style: .destructive, handler: { [weak self] _ in
+            guard let self = self else { return }
+            
+            self.delegate?.deleteItem(item: item, at: indexPath.row)
+            
+            if self.delegate?.currentPlaylistItem == nil {
+                self.updateTableBackgroundView()
+                self.activityIndicator.stopAnimating()
+            }
+        }))
+        
+        alert.addAction(UIAlertAction(title: Strings.cancelButtonTitle, style: .cancel, handler: nil))
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func openInNewTab(_ item: PlaylistInfo, isPrivate: Bool) {
+        if let browser = PlaylistCarplayManager.shared.browserController,
```

Comment:
This is old code. But it is because we need to sync what is happening on the phone with what is happening in the car.
If a user selects a song on the phone, it needs to push the same song's UI on the car.
If a user selects a song on the car, it needs to push the same song's UI on the phone.

The requirements for this and the logic for it is insanely complicated. IMHO, the user should not be using the phone in the car at all. But someone pointed out that a passenger can do it and it should affect each other.